src/roman_datamodels/datamodels/_utils.py

```diff
@@ -302,3 +319,64 @@ def rdm_open(init, memmap=False, **kwargs):
 
         asdf_file.close()
         raise TypeError(f"Unknown datamodel type: {model_type}")
+
+
+def rdm_open(init, memmap=False, *, auto_upgrade=True, **kwargs):
+    """
+    Datamodel open/create function.
+        This function opens a Roman datamodel from an asdf file or generates
+        the datamodel from an existing one.
+
+    Parameters
+    ----------
+    init : str, ``Path``, `DataModel`, `asdf.AsdfFile`, file-like
+        May be any one of the following types:
+            - `asdf.AsdfFile` instance
+            - string or ``Path`` indicating the path to an ASDF file
+            - `DataModel` Roman data model instance
+            - file-like object compatible with `asdf.open`
+    memmap : bool
+        Open ASDF file binary data using memmap (default: False)
+
+    auto_upgrade : bool
+        If True, automatically upgrade the model to the latest version if needed.
+
+    Returns
+    -------
+    `DataModel`
+    """
+    output = _rdm_open(init, memmap=memmap, **kwargs)
+
+    # Bail out if not a datamodel
+    if not isinstance(output, DataModel):
+        return output
+
+    # Bail out if upgrade is disabled
+    if not auto_upgrade:
+        return output
+
+    # Bail out if read tag is found so it is already the latest version
+    if (tag := output._instance._read_tag) is None:
+        return output
+
+    # Bail out if the tag is not in the internal tag removal map
+    #   We may want to consider removing this check and always upgrading
+    if tag not in _INTERNAL_TAG_REMOVAL_MAP:
+        return output
+
+    warnings.warn(
+        f"Converting {tag} to structually equivalent {_INTERNAL_TAG_REMOVAL_MAP[tag]} with no internally tagged nodes",
+        RadConversionWarning,
+        stacklevel=2,
+    )
+    output.validate()
+
+    def _callback(node):
+        if isinstance(node, TaggedScalarNode):
+            return node.unwrap()
+        return node
+
+    new = type(output).create_from_model(asdf.treeutil.walk_and_modify(dict(output._instance), _callback))
+    new._instance._read_tag = _INTERNAL_TAG_REMOVAL_MAP[tag]
```

Comment:
```suggestion
    output.validate()

    def _callback(node):
        if isinstance(node, TaggedScalarNode):
            return node.unwrap()
        return node

    new = type(output).create_from_model(asdf.treeutil.walk_and_modify(dict(output._instance), _callback))
    new._instance._read_tag = _INTERNAL_TAG_REMOVAL_MAP[tag]
```
Would you split the `auto_upgrade` feature into a new PR? There are a few issues here:
- should the default be `True`?
- should the upgraded model be validated?
- how can this be done efficiently? (the current code reads the entire input model into memory, creates a copy via walk_and_modify, creates a second copy via create_from_model)

I'd also like to see what romancal tests fail without this auto_upgrade. Can these all be addressed by regenerating regtest files?